src/components/pages/Masters/StateDetails.js

```diff
@@ -0,0 +1,126 @@
+    const [form, setForm] = useState({});
+    const [title, setTitle] = useState();
+
+    const { updateState } = useUpdateState(() => {
+        toast.success(`${ form.values.name} updated successsfully.`);
+        onSubmit();
+    }, errorCallback);
+
+    const { createState } = useCreateState((response) => {
+        if (response instanceof AxiosError) {
+            errorCallback();
+        } else {
+            toast.success(`${form.values.name} created successsfully.`);
+            onSubmit();
+        }
+    }, errorCallback);
+
+    function errorCallback() {
+        toast.error(ERROR_MESSAGES.DEFAULT);
+    }
+    const schema = {
+        fields: [
+            {
+                component: action === ACTIONS.VIEW ? componentTypes.PLAIN_TEXT : componentTypes.TEXT_FIELD,
+                name: 'name',
+                label: 'State Name',
+                validate: [
+                    { type: validatorTypes.REQUIRED }
+                ],
+                content: getValue(data, 'name')
+            },
+            {
+                component: action === ACTIONS.VIEW ? componentTypes.PLAIN_TEXT : componentTypes.TEXT_FIELD,
+                name: 'code',
+                label: 'State Code',
+                validate: [
+                    { type: validatorTypes.REQUIRED },
+                    { type: validatorTypes.MAX_LENGTH, threshold: 2 },
+                    { type: validatorTypes.PATTERN, pattern: /[a-zA-Z]{2}/, message: 'Should be alpha value of length 2' }
+                ],
+                styleClass: 'text-uppercase',
+                content: getValue(data, 'code')
+                
+            }   
+        ],
+    };
+
+    function submitState(e) {
+        preventDefault(e);
+        if (form.valid) {
+            const { code, name } = form.values;
+            const payload = {
+                code: code.toUpperCase(),
+                name,
+            };
+            if (action === ACTIONS.EDIT) {
+                payload['id'] = data.id;
+                updateState(payload)
+            } else if (action === ACTIONS.ADD) {
+                createState(payload);
+            }
+        }
+    }
+
+    useEffect(() => {
+        if (action) {
+            switch (action) {
+                case ACTIONS.ADD:
+                    setTitle('Add State Master');
+                    break;
+                case ACTIONS.EDIT:
+                    setTitle('Edit State Master');
+                    break;
+                case ACTIONS.VIEW:
+                    setTitle('View State Master');
+                    break;
+                default:
+                    setTitle('State Master');
+            }
+        }
+    }, [action]);
+
+    return (
+        <>
+            <Modal show={true} backdrop="static" dialogClassName="drawer" animation={false}>
+                <Modal.Header closeButton={true} onHide={onClose}>
+                    <Modal.Title className="bg">{title}</Modal.Title>
```

Comment:
call GetActionTitle('Law') from master.constants.js for title and remove line#78-94